code/modules/reagents/reagent_containers.dm

```diff
@@ -24,6 +24,13 @@
 	if(list_reagents)
 		reagents.add_reagent_list(list_reagents)
 
+/obj/item/weapon/reagent_containers/CheckParts(list/content)
+	..()
+	var/datum/reagents/RS = new(volume)

```

Comment:
same thing, reagent_containers/New() doesn't handle reagents and volume creation?